File: frontend/src/types/profiles.ts
export interface ProfileDefinition {
  id: string;
  display_name: string;
  delta_level: string;
  theta_level: string;
  alpha_level: string;
  beta_level: string;
  gamma_level: string;
  notes?: string | null;
}

export interface ProfileSet {
  id: string;
  name: string;
  description: string;
  profiles: ProfileDefinition[];
}

export interface ProfileSetSummary {
  id: string;
  name: string;
  description: string;
  profile_count: number;
}

export const WAVE_LEVELS = ['yüksek', 'yüksek orta', 'orta', 'düşük orta', 'düşük'] as const;
export type WaveLevel = typeof WAVE_LEVELS[number];

File: backend/app/models/profiles.py
from pydantic import BaseModel
from typing import List


class ProfileDefinition(BaseModel):
    id: str  # e.g. "YARATICI_LIDER"
    display_name: str  # "YARATICI LİDER"
    delta_level: str  # "yüksek", "orta", etc.
    theta_level: str
    alpha_level: str
    beta_level: str
    gamma_level: str
    notes: str | None = None


class ProfileSet(BaseModel):
    id: str  # "meditasyon"
    name: str  # "Meditasyon Profilleri"
    description: str = ""
    profiles: List[ProfileDefinition]


class ProfileSetSummary(BaseModel):
    id: str
    name: str
    description: str = ""
    profile_count: int
